### ISSUE TITLE
Orchestration function getting invoked after every Activity completion even on Task.WhenAll

### ISSUE THREAD

**Issue by raghavendra3790** [Azure/azure-functions-durable-extension#92]
I am not sure if this is the intended behavior, but here is the issue.

I have a fan out scenario. The Orchestration function starts few Activity functions and waits for all of them to complete. But the Orchestration function is getting triggered every time an Activity function is completed. This is increasing the function execution times exponentially. I would expect the Orchestration function to wake up only after all the tasks completed because I said Task.WhenAll. 

For instance if I have 10 activity functions invoked in fan out manner. This means 10 more invocations of my orchestration function. It would be great if we can avoid this.

**Comment by cgillum** [comment]
Thanks for opening this issue. The behavior is mostly by design. However, depending of the timing of your activity completions, you *should* observe that as many as 32 concurrent completions can be handled in a single precursor replay. It's not possible to avoid all replays entirely.

That said, work needs to be done to improve throughput in this scenario. There is a separate issue tracking that: https://github.com/Azure/azure-functions-durable-extension/issues/81. Hopefully that will help relieve your concern. Let me know if you have further questions related to this.
